### PR TITLE
#170947574 supplier add an accommodation

### PR DESCRIPTION
**Pull request by izzett222**
#### What does this PR do?
allow an accommodation supplier to add a facility into the system
#### Description of Task to be completed?
- changed the add facility route to authorize a supplier
- add the supplier as user role
- add supplier add facility feature test
- change the route name to follow convention
- modify the facility swagger documentation
- create the facility controller and route
- add createBy, image, amenities, and services attribute to the Facilities table
#### How should this be manually tested?
- clone this repo
- create the .env file and populate it with the environment variable in env.exampe
- run npm install
- setup the development database and the test database
- run npm reset: dev
- open postman and make a post to /api/v1/facilities
- run npm run test to test the app

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
#170947574
#### Screenshots (if appropriate)
Screen Shot 2020-02-25 at 13 34 31
Screen Shot 2020-02-25 at 13 34 01
Screen Shot 2020-02-25 at 13 22 19

### CODE REVIEW

**Review by bbaime98 — DISMISSED** [review]

**Review by Ntare22 — DISMISSED** [review]
This looks good.. great work @izzett222....

File: src/controllers/facilitiesController.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    * @param {Object} res
+    * @returns {Object} Room
+    * @memberof FacilitiesController
+  */
+  static async createRoom(req, res) {
+    try {
+      const { facilityId, roomName, type } = req.body;
+      const existingFacility = await db.Facilities.findOne({ where: { id: facilityId } });
+      if (!existingFacility) {
+        return Response.errorResponse(res, 401, 'This facility does not exist');
```
Comment (Baraka-Mugisha) [inline src/controllers/facilitiesController.js]: We should internationalize this too

Comment (izzett222) [reply src/controllers/facilitiesController.js]: let me work on it

**Review by Mnickii — DISMISSED** [review]

File: src/controllers/facilitiesController.js (hunk 1 above)
Comment (izzett222) [reply src/controllers/facilitiesController.js]: I added internalization that respose

**Review by Baraka-Mugisha — APPROVED** [review]

**Review by Ntare22 — APPROVED** [review]
Great work creating a facilities router... I think It add more organisation and clarity. LGTM

**Review by bbaime98 — APPROVED** [review]

**Pull request merged**